src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESIT.java

```diff
@@ -0,0 +1,275 @@
+                    .must(matchQuery(caseData(TEXT_FIELD), TEXT_VALUE)) // ES Text
+                    .must(matchQuery(caseData(DATE_FIELD), DATE_VALUE)) // ES Date
+                    .must(matchQuery(caseData(PHONE_FIELD), PHONE_VALUE)) // ES Phone
+                    .must(matchQuery(caseData(COUNTRY_FIELD), COUNTRY_VALUE)) // Complex
+                    .must(matchQuery(caseData(COLLECTION_FIELD) + VALUE_SUFFIX, COLLECTION_VALUE)) // Collection
+                    .must(matchQuery(STATE, STATE_VALUE)))
+                .build().toJsonString();
+
+            MvcResult result = mockMvc.perform(post(POST_SEARCH_CASES)
+                .contentType(MediaType.APPLICATION_JSON)
+                .param(CASE_TYPE_ID_PARAM, CASE_TYPE_A)
+                .content(searchRequest))
+                .andExpect(status().is(200))
+                .andReturn();
+
+            String responseAsString = result.getResponse().getContentAsString();
+            CaseSearchResult caseSearchResult = mapper.readValue(responseAsString, CaseSearchResult.class);
+
+            CaseDetails caseDetails = caseSearchResult.getCases().get(0);
+            assertAll(
+                () -> assertThat(caseSearchResult.getTotal(), is(1L)),
+                () -> assertExampleCaseMetadata(caseDetails),
+                () -> assertExampleCaseData(caseDetails)
+            );
+        }
+
+        @Test
+        void shouldErrorWhenInvalidCaseTypeIsProvided() throws Exception {
+            String searchRequest = ElasticsearchTestRequest.builder()
+                .query(matchAllQuery())
+                .build().toJsonString();
+
+            MvcResult result = mockMvc.perform(post(POST_SEARCH_CASES)
+                .contentType(MediaType.APPLICATION_JSON)
+                .param(CASE_TYPE_ID_PARAM, "INVALID")
+                .content(searchRequest))
+                .andExpect(status().is(404))
+                .andReturn();
+
+            String responseAsString = result.getResponse().getContentAsString();
+            JsonNode exceptionNode = mapper.readTree(responseAsString);
+
+            assertAll(
+                () -> assertThat(exceptionNode.get("message").asText(),
+                    startsWith("Resource not found when getting case type definition for INVALID"))
+            );
+        }
+
+        public void assertExampleCaseMetadata(CaseDetails caseDetails) {
+            assertAll(
+                () -> assertThat(caseDetails.getJurisdiction(), is("AUTOTEST1")),
+                () -> assertThat(caseDetails.getCaseTypeId(), is(CASE_TYPE_A)),
+                () -> assertThat(caseDetails.getCreatedDate().toString(), is("2020-05-07T15:53:40.974")),
+                () -> assertThat(caseDetails.getLastModified().toString(), is("2020-06-09T13:17:06.542")),
+                // () -> assertThat(caseDetails.getLastStateModifiedDate().toString(), is("TBC")), // TODO: After RDM-8552 available
```

Comment:
This should be available now so we should uncomment this line

Reply:
Merged in latest and uncommented.